tests/rbd/rbd_utils.py

```diff
@@ -730,6 +743,79 @@ def set_config(self, level, entity, key, value, **kw):
         """
         return self.exec_cmd(cmd=f"rbd config {level} set {entity} {key} {value}", **kw)
 
+    def encrypt(self, image_spec, encryption_type, passphrase):
+        """Apply luks encryption of the specified format on the given image
+
+        Args:
+            image_spec (str): pool/image on which encryption to be applied
+            encryption_type (str): luks1, luks2
+            passphrase (str): path to passphrase file
+
+        Returns:
+            int: 0 if encryption successful else 1
+        """
+        cmd = f"rbd encryption format {image_spec} {encryption_type} {passphrase}"
+        return self.exec_cmd(cmd=cmd)
+
+    def device_map(self, image_spec, device_type, encryption_config):
+        """Perform rbd device map operation using given parameters
+
+        Args:
+            image_spec (str): pool_name/image_name
+            device_type (str):  arg device type [ggate, krbd (default), nbd]
+            encryption_config (list(dict)): ["<encryption_type>": "<corresponding_passphrase>"]
+
+        Returns:
+            Status of rbd device map command execution (1 if failed, 0 if passed)
+        """
+        cmd = f"rbd device map -t {device_type} "
+        if encryption_config:
+            cmd += "-o "
+            for encryption in encryption_config:
+                for format, phrase in encryption.items():
+                    cmd += f"encryption-format={format},encryption-passphrase-file={phrase},"
+            cmd = cmd[:-1]  # remove trailing comma
+        cmd += f" {image_spec}"
+        # add validations to return value as per document
+        return self.exec_cmd(cmd=cmd, all=True)
+
+    def device_unmap(self, image_spec, device_type, encryption_config):
+        """Perform rbd device unmap operation using given parameters
+
+        Args:
+            image_spec (str): pool_name/image_name
+            device_type (str):  arg device type [ggate, krbd (default), nbd]
+            encryption_config (list(dict)): ["<encryption_type>": "<corresponding_passphrase>"]
+
+        Returns:
+            Status of rbd device unmap command execution (1 if failed, 0 if passed)
+        """
+        cmd = f"rbd device unmap -t {device_type} "
+        if encryption_config:
+            cmd += "-o "
+            for encryption in encryption_config:
+                for format, phrase in encryption.items():
+                    cmd += f"encryption-format={format},encryption-passphrase-file={phrase},"
+            cmd = cmd[:-1]  # remove trailing comma
+        cmd += f" {image_spec}"
+        # add validations to return value as per document
+        return self.exec_cmd(cmd=cmd)
```

Comment:
can we combine this two method by passing extra parameter action as "map" & "unmap"

Reply:
Done.